Remove unneeded property

evpobr opened the pull request:
`POSITION_INDEPENDENT_CODE` is [True by default](https://cmake.org/cmake/help/v3.1/prop_tgt/POSITION_INDEPENDENT_CODE.html).

evpobr committed:
Remove unneeded property

syb0rg commented:
> This property is True by default for SHARED and MODULE library targets and False otherwise.

Your commit removes the property for the static library as well, which isn't true by default.

evpobr commented:
@syb0rg PIC makes sense for [shared libraries only](http://www.tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html#AEN95).

evpobr commented:
@syb0rg Good god, seems like we made it all wrong. x64 build may work, need to test under 32-bit Linux.

I guess now x64 static and shared Linux libraries allways use some kind of PIC, but what about 32-bit libraries? Seems like when no PIC property set, we will try to link non-PIC static library to PIC-enabled shared library.

And if PIC is set manually, we build correct shared library and useless PIC-enabled static library?

@syb0rg , you can test CMake build under 32-bit Linux?

@erikd, can you help? May be we need not to link static library to shared, but build them separately, to allow CMake provide correct compiler flags?

evpobr commented:
@syb0rg automake build creates two object file for each c file, one in src/ and another in src/.libs. We must build static and shared library each time from sources.

syb0rg commented:
@evpobr I don't have a 32-bit Linux machine.  I may be able to emulate one however.

We already build them separately in CMake... I'm not sure what you're getting at.

evpobr commented:
@syb0rg, automake build provides PIC-disabled static library and PIC-enabled shared library. That's how it used to be. So automake script builds sources two times, with PIC and without it.

We build all sources once with PIC flag to create static library and link it against PIC shared library. Our shared library is OK, but static library is not.

The way to make things right is not to link static library against shared. We must first build static library from sources without PIC, and then build shared library from sources with PIC.

syb0rg commented:
Why can't we just set the static library proprieties/PIC options to false?

evpobr commented:
It's already false by default. We get PIC-disabled static library and and we will try to link it against PIC-enabled shared library. Now we have static library OK, but what happens when we link it to shared one? It seems to work under x64, but under x32? I guess it will not even compile. Testing needed.

evpobr commented:
@syb0rg Look at answer in [this thread](http://stackoverflow.com/questions/8808691/can-i-build-a-shared-library-by-linking-static-libraries)

syb0rg commented:
So then why don't we just make the static library have PIC like we were doing?

evpobr commented:
@syb0rg Wait for commit.

evpobr committed:
Remove sndfile_static target

sndfile_static removed. You can tell CMake what to build with
BUILD_SHARED_LIBS option.  Why?

Pros:

* CMake or you may want to provide dirrerent compiler options for static
or shared build. Now you can do it. For example, CMake will add
POSITION_INDEPENDENT_CODE to shared build, and set it off for static
build to make code faster.
* You can select, what version to build.
* You can control to link programs and tests against static or shared
library.
* Some platforms do not support static or shared libraries. CMake will
select appropriate build for you.

Cons:

* You must run cmake and build targets twice if you want to get both
static and shared library.

erikd commented on CMakeLists.txt:
```diff
@@ -6,6 +6,7 @@
 # Options
+option (BUILD_SHARED_LIBS "Build shared libraries" OFF)
 option (ENABLE_TESTING "Enable testing" OFF)
 
```
I actually think that should default to `ON`.

evpobr commented:
@erikd `BUILD_SHARED_LIBS` option is just a configure helper for system variable, which is not set by default. It's CMake's behavior, and user expects to see static library, if `BUILD_SHARED_LIBS` is not set manually.

Also It's platform specific. Under Linux shared libraries are preffered, under Windows it's safer to link static libs because of DLL hell.
